relevanceai/dataset/write/write.py

```diff
@@ -952,3 +1008,35 @@ def insert_list(self, labels: list, label_field: str = "label", **kwargs):
         """
         documents = [{label_field: l} for l in labels]
         return self.insert_documents(documents=documents, **kwargs)
+
+    def batched_upsert_media(
+        self,
+        images: List[str],
+        show_progress_bar: bool = False,
+    ) -> List[str]:
+        """batched upsert media"""
+
+        bs = int(len(images) / (os.cpu_count() + 4))
+        nb = int(len(images) / bs)
+
+        with concurrent.futures.ThreadPoolExecutor() as executor:
```

Comment:
Need to make number of threads adjustable as this can either crash the system or be severely underused.